### PR TITLE
CORE-427: In System.initialize() don't filter on `networksNeeded`

### PR DESCRIPTION
**Pull request by EBGToo**
Rename System.start to System.initialize.

### CODE REVIEW

File: Swift/BRCrypto/BRCryptoSystem.swift [hunk 1]
```diff
@@ -34,6 +34,99 @@ public final class System {
...
 
+    ///
+    /// Return the AddressSchemes support for `network`
+    ///
+    /// - Parameter network: the network
+    ///
+    /// - Returns: An array of AddressScheme
+    ///
+    public func supportedAddressSchemes (network: Network) -> [AddressScheme] {
+        switch network.currency.code {
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoSystem.swift]: The Law of Demeter gently weeps...

Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoSystem.swift]: Hah, I feel younger now; remember past discussions, years ago.

File: Swift/BRCryptoDemo/CoreDemoAppDelegate.swift [hunk 2]
```diff
@@ -135,7 +143,7 @@ class CoreDemoAppDelegate: UIResponder, UIApplicationDelegate, UISplitViewContro
     func applicationWillEnterForeground(_ application: UIApplication) {
         // Called as part of the transition from the background to the active state; here you can
         // undo many of the changes made on entering the background.
-        system.start(networksNeeded: [])
+        system.managers.forEach { $0.connect() }
```
Comment (michael-brd) [inline Swift/BRCryptoDemo/CoreDemoAppDelegate.swift]: Is it weird that we have a stop but no start?

Comment (EBGToo) [reply Swift/BRCryptoDemo/CoreDemoAppDelegate.swift]: Well, System.stop() can safely call disconnect() on every WalletManager, connected or not.  But, yes.  Let the App handle it?

File: Swift/BRCrypto/BRCryptoSystem.swift (hunk 1 above)
Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoSystem.swift]: Maybe we have a chance to remove `Currency.codeAs{BTC,...}`?

File: Swift/BRCrypto/BRCryptoSystem.swift [hunk 3]
```diff
@@ -158,22 +252,14 @@ public final class System {
...
+    /// @Note: This should only be called one.
     ///
-    public func start (networksNeeded: [String]) {
-        if !networks.isEmpty {
-            managers.forEach { $0.connect() }
-            return
-        }
+    public func configure () {
 
         #if MAINNET
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoSystem.swift]: if MAINNET and TESTNET have cropped up more and more. This paradigm doesn't map to other languages. Would be ideal if we had this parameterized wherever we are using this pattern.

File: Swift/BRCrypto/BRCryptoWalletManager.swift [hunk 4]
```diff
@@ -165,9 +162,7 @@ public final class WalletManager: Equatable {
         self.query   = system.query
 
         self.defaultNetworkFee = network.minimumFee
-
-        self.addressSchemes = WalletManager.defaultAddressSchemes (currency: network.currency)
-        self.addressScheme  = AddressScheme (core: cryptoWalletManagerGetAddressScheme (core))
+        self.addressScheme     = AddressScheme (core: cryptoWalletManagerGetAddressScheme (core))
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoWalletManager.swift]: Is this needed? Is the getter sufficient?

File: Swift/BRCryptoDemo/CoreDemoAppDelegate.swift [hunk 5]
```diff
@@ -31,6 +31,14 @@ class CoreDemoAppDelegate: UIResponder, UIApplicationDelegate, UISplitViewContro
     var listener: CoreDemoListener!
     var system: System!
 
+    #if TESTNET
```
Comment (michael-brd) [inline Swift/BRCryptoDemo/CoreDemoAppDelegate.swift]: Same comment re:defines

File: Swift/BRCryptoDemo/CoreDemoAppDelegate.swift [hunk 6]
```diff
@@ -86,9 +94,20 @@ class CoreDemoAppDelegate: UIResponder, UIApplicationDelegate, UISplitViewContro
         print ("APP: Account PaperKey  : \(paperKey.components(separatedBy: CharacterSet.whitespaces).first ?? "<missed>") ...")
         print ("APP: Account Timestamp : \(account.timestamp)")
         print ("APP: StoragePath       : \(storagePath)");
+        print ("APP: Mainnet           : \(mainnet)")
+        var currencies: [String] = ["btc", "eth", "brd" /*, "xrp"*/]
+
+        if mainnet {
```
Comment (michael-brd) [inline Swift/BRCryptoDemo/CoreDemoAppDelegate.swift]: Left in unintentionally?

File: Swift/BRCrypto/BRCryptoSystem.swift (hunk 3 above)
Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoSystem.swift]: We have CORE-284 coming, from Aaron, that eliminates the compiler flag altogether.  BRChainParams changes to include the differences in BITCOIN_TESTNET

File: Swift/BRCryptoDemo/CoreDemoAppDelegate.swift [hunk 7]
```diff
@@ -105,18 +124,7 @@ class CoreDemoAppDelegate: UIResponder, UIApplicationDelegate, UISplitViewContro
...
-        networksNeeded += ["bitcoin-testnet"] //, "ripple-testnet"] // ...
-        #endif
-
-        #if MAINNET
-        networksNeeded += ["bitcoin-mainnet"] // ...
-        #endif
-
-        print ("APP: Networks          : \(networksNeeded)")
-        self.system.start (networksNeeded: networksNeeded) //, */"ethereum-mainnet"])
+        self.system.configure()
```
Comment (michael-brd) [inline Swift/BRCryptoDemo/CoreDemoAppDelegate.swift]: Food for thought, but do we want to enforce that configure has been run once (and only once)?

File: Swift/BRCryptoDemo/CoreDemoAppDelegate.swift (hunk 6 above)
Comment (EBGToo) [reply Swift/BRCryptoDemo/CoreDemoAppDelegate.swift]: Yes.

File: Swift/BRCrypto/BRCryptoSystem.swift (hunk 3 above)
Comment (michael-brd) [reply Swift/BRCrypto/BRCryptoSystem.swift]: We still will be providing if it is mainnet to BDB, right?

**Pull request merged**